lib/archethic/contracts/constants.ex

```diff
@@ -197,4 +199,21 @@ defmodule Archethic.Contracts.Constants do
 
     Map.put(constants, "balance", balance_constants)
   end
+
+  @doc """
+  Add the genesis address of the transaction
+  """
+  @spec resolve_genesis_address(map()) :: map()
+  def resolve_genesis_address(constants) do
+    previous_address =
+      constants
+      |> Map.get("previous_public_key")
+      |> Base.decode16!()
+      |> Crypto.derive_address()
+
+    nodes = Election.chain_storage_nodes(previous_address, P2P.authorized_and_available_nodes())
+    {:ok, genesis_address} = TransactionChain.fetch_genesis_address(previous_address, nodes)
```

Comment:
While it's doing the job, I think there is some I/O which could be removed.
For instance as the genesis address is resolved in each transaction's validation, we could send the genesis address already fetched into the interpreter avoiding this request to be made again, saving some I/O.